frontend/catalyst/third_party/oqd/src/CMakeLists.txt

```
@@ -26,6 +26,8 @@ set(OQD_LIBRARIES
 set_target_properties(rtd_oqd PROPERTIES BUILD_RPATH "$ORIGIN/../utils")
 target_link_directories(rtd_oqd PRIVATE ${runtime_lib})
 
+target_link_libraries(rtd_oqd PRIVATE ${runtime_lib}/librt_OQD_capi.so)
```

These would fail for a Darwin system that uses .dylib. 
```suggestion
if(APPLE)
    target_link_libraries(rtd_oqd PRIVATE ${runtime_lib}/librt_OQD_capi.dylib)
 elseif(UNIX)
     target_link_libraries(rtd_oqd PRIVATE ${runtime_lib}/librt_OQD_capi.so)
endif()
```

---

There's actually no need to explicitly link to the actual .so file. We should just link to a cmake target and let cmake do the magic itself, i.e. we should just 
```
target_link_libraries(rtd_oqd_device PRIVATE rt_OQD_capi)
```

(This explicit link was my hack, since the `rt_OQD_capi` target was in the runtime but this cmake file itself is in frontend. We no longer need this hack now that oqd is entirely in runtime!)